src/lib/stores/canvas.svelte.ts

```
@@ -267,6 +268,67 @@ function fitAll(racks: Rack[], rackGroups: RackGroup[] = []): void {
+
+  // Find groups that contain ANY of the target racks
+  // If focusing on a rack in a bayed group, we focus on the entire group
+  const targetRackIdSet = new Set(rackIds);
+  const relevantGroups = rackGroups.filter(
+    (g) =>
+      g.layout_preset === "bayed" &&
+      g.rack_ids.some((id) => targetRackIdSet.has(id)),
+  );
+
+  // Collect all racks we need to focus on (including all racks in relevant bayed groups)
+  const allRelevantRackIds = new Set(rackIds);
```

ESLint

[error] Found a mutable instance of the built-in Set class. Use SvelteSet instead.

(svelte/prefer-svelte-reactivity)